iib/workers/tasks/fbc_utils.py

```diff
@@ -17,6 +19,45 @@ def is_image_fbc(image):
     :return: True if image is FBC type, False otherwise (SQLite)
     :rtype: bool
     """
+    from iib.workers.tasks.utils import skopeo_inspect
+
     skopeo_output = skopeo_inspect(f'docker://{image}')
-    dc_image_label = 'operators.operatorframework.io.index.configs.v1'
-    return dc_image_label in skopeo_output['Labels']
+    fbc_image_label = 'operators.operatorframework.io.index.configs.v1'
+    return fbc_image_label in skopeo_output['Labels']
+
+
+def get_index_fbc_dir(from_index, base_dir):
+    """
+    Get file-based catalog directory from the specified index image and save it locally.
+
+    :param str from_index: index image to get file-based catalog directory from.
+    :param str base_dir: base directory to which the database file should be saved.
+    :return: path to the copied file-based catalog directory.
+    :rtype: str
+    :raises IIBError: if any podman command fails.
+    """
+    from iib.workers.tasks.build import _copy_files_from_image
+    from iib.workers.tasks.utils import get_image_label
+
+    fbc_dir_path = get_image_label(from_index, 'operators.operatorframework.io.index.configs.v1')
+    if not fbc_dir_path:
+        raise IIBError('Index image doesn\'t have the label specifying its catalog location.')
+    _copy_files_from_image(from_index, fbc_dir_path, base_dir)
+    return os.path.join(base_dir, os.path.dirname(fbc_dir_path))
+
+
+def get_hidden_index_database(from_index, base_dir):
```

Comment:
Would it be possible to merge functions **get_hidden_index_database** and **get_index_fbc_dir** then seem to have similar use case. Copy and join the path are used in both of them.

Reply:
I do not see a nice way how to merge it. Do you have something specific in mind?

Reply:
Could it be a part of the code where you call the get_hidden_index_database?

Reply:
IIUC `get_index_fbc_dir` returns the catalog path for FBC images and `get_hidden_index_database` returns the DB path for legacy images.

IMO it's possible to merge them in a single function, whatever the name it is by using a single and generic *whatever_path* variable, more or less like this:

1. Call `is_image_fbc(from_index)` to verify if the image is FBC or not
2. **If** FBC do try to get `whatever_path` using `get_image_label(from_index,  'operators.operatorframework.io.index.configs.v1')` and verify if it's set, raising IIBError if not, as it's implemented; **else**  try to get path from `get_worker_config().get('hidden_index_db_path')` and verify if it's set or raise `IIBError` due to misconfiguration
3. Once you have your generic `whatever_path` you use `_copy_files_from_image` replacing `fcb_dir_path` and `db_path` with this generic one obtained in step 2
4. Finally return the `os.path.join(base_dir, os.path.basename(whatever_path))`

**Note:** The variable `whatever_path` should have a nice name, so here I'm just using this name to exemplify how I imagine these 2 functions can be merged

Reply:
Also I don't see the function `get_hidden_index_database` being used anywhere in this PR... So I believe it can be merged, but if you do that please rename `get_index_fbc_dir` to something more generic, like `get_index_catalog_dir` and remember to change its calls on `iib.workers.tasks.utils` to match the new name

Reply:
@lipoja please let me know the use `get_hidden_index_dabase` since I don't see calling for it anywhere and also whether you consider the merge with `get_index_fbc_dir`

Reply:
@pbortlov @JAVGan Personally it does not feel right to me to merge them. I wanted to have simple way how to get hidden index.db (and path to it) and also similar to get path to file-based catalog directory copied from index image. 
If I merge it then the function is more complicated and the caller has to specify what should be the output - if you want index.db or catalog directory. 

And also please have a look now - I was fixing the code and both functions have now common only calling `_copy_files_from_image` . The rest is changed. Therefore I would prefer to keep them separate.

Reply:
thanks Jan for the work, explanation and changes :)